[Adee] iP

adeearyaa opened the pull request:
### DukeBeast

“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

**DukeBeast frees your mind of having to remember things you need to do. It's,**

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

**download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).**

1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is FREE!

Features:

 

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
``` ruby
public class Main {
    public static void [] main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

adeearyaa committed:
Level-1

adeearyaa committed:
Level-2

adeearyaa committed:
Level-3

adeearyaa committed:
Level-4

adeearyaa committed:
Level 4 v2

adeearyaa committed:
Level-4 v3

adeearyaa committed:
A-TextUITesting

adeearyaa committed:
Level-5

adeearyaa committed:
Level-6

adeearyaa committed:
Level -6

adeearyaa committed:
Level-7

adeearyaa committed:
Level-8

adeearyaa committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/DukeOperations.java

adeearyaa committed:
Level -8

adeearyaa committed:
A-MoreOOP

adeearyaa committed:
A-Packages

adeearyaa committed:
A-Packages

adeearyaa committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
adding gradle to the previous files

adeearyaa committed:
A-Gradle

adeearyaa committed:
A-JUNit

adeearyaa committed:
Adding of Javadocs to half the classes and methods

adeearyaa committed:
Following java code standards

adeearyaa committed:
Allowing of users to search for the tasks using keywords

adeearyaa committed:
Merge branch 'branch-Level-9'

adeearyaa committed:
Some changes to the data storage system in the file

bokun2 commented on src/main/java/duke/DukeOperations.java:
```diff
@@ -0,0 +1,92 @@
+    * method that returns exit message.
+    *
+    */
+    public void exit() {
```
Perhaps can leave all the interactive sentences into a Ui class?  🤔 This may make the code easier to read. For example, a ui.showWelcomeMessage() method.

bokun2 commented on src/main/java/duke/DukeOperations.java:
```diff
@@ -0,0 +1,92 @@
+     * @param array arraylist of all the tasks so far.
+     * @param number integer value of the task to be unmarked as not done.
+     * */
+     public void unmark(ArrayList<Task>array, int number) {
```
For method naming, perhaps you can use unMark instead of unmark. I also noticed similar issues in other methods.

bokun2 commented on src/main/java/duke/DukeOperations.java:
```diff
@@ -0,0 +1,92 @@
+     public void findWord(String word, ArrayList<Task> array, int counter) {
+         int j = 0;
+         for (int i = 0; i < counter; i++) {
+             if (array.get(i) != null && array.get(i).toString().contains(word)) {
```
This sentence is long and a little bit hard for reading. Since array.get(i).toString()appears more than once, maybe you can change it into a variable like taskName?

bokun2 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+     @Override
+     public String toString() {
+         LocalDate d1 = LocalDate.parse(this.date);
+         return "[E]" + "[" + super.getStatusIcon() + "]" +  super.toString() + " (at: " + d1.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + " " + this.timing +  ")";
```
Also the return statement is a little hard to read.

bokun2 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+                ops.displayList(array, counter);
+
+            } else if (firstword.equals("todo")) {
+                if (!storage.fileExists()) {
```
Perhaps this part can be modified a little so it is no longer nested?

bokun2 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+        System.out.println("Hello! I'm Duke What can I do for you?");
+        ArrayList<Task> array = new ArrayList<>(100);
+        int counter = 0;
+        boolean flag = false;
```
For naming, perhaps this variable can be named as isXX because it is a boolean value. 🤔

bokun2 reviewed (commented):
The code is clean to read.

adeearyaa committed:
GUI

adeearyaa committed:
Adding Assertions to Duke

In this commit a few assertion shave been added to check which command has been entered by the user

adeearyaa committed:
ignore

adeearyaa committed:
Improving the current code quality of Duke

-changing names of variables
-adding more javadocs

adeearyaa committed:
Merge pull request #1 from adeearyaa/branch-A-CodeQuality

Improving the current code quality of Duke

adeearyaa committed:
Merge branch 'master' into branch-Assertions

adeearyaa committed:
Now duke is able to detect duplicates as well

adeearyaa committed:
Merge branch 'branch-C-DetectDuplicates'

adeearyaa committed:
Some changes to the naming of methods has been done

adeearyaa committed:
Made some changes to the product website and also added a user guide

adeearyaa committed:
some extra changes

adeearyaa committed:
extra changes

adeearyaa committed:
Fix comments and some minor bugs and code quality issues
